compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/window/Dialog.skiko.kt

```diff
@@ -193,6 +194,13 @@ private fun DialogLayout(
     }
 }
 
+@Composable
+private fun rememberLayerContent(layer: ComposeSceneLayer, content: @Composable () -> Unit) {
+    remember(layer, content) {
+        layer.setContent(content)
```

Comment:
This destroys the previous composition, and creates a new one.

I think, we should do only a recomposition, as we do with usual Composable functions that accept `content`:
```
rememberComposeSceneLayer(content)

// rememberComposeSceneLayer
val currentContent by rememberUpdateState()

val layer = remember {
    scene.createLayer(...)
    scene.setContent {
         currentContent()
    }
}
```

But if we still want to recreate the composition, the current code is fine.

Reply:
> This destroys the previous composition, and creates a new one.

Sure, it's the reason of this PR - it recreates it each recomposition before.

> But if we still want to recreate the composition

We should set (recreate) it if the LEAYER is changed, so the new layer don't have content at all. For the content it's the same - I don't see any reason to introduce such caching for elements that produce a new layer and should behave the same as `setContent` in the root entry point

Reply:
> should behave the same as setContent in the root entry point

The reason I suggested it - users of `Dialog` and `Popup` don't see the `setContent` function, they see the usual Composable API:

```
fun MyComposable() {
  Dialog(content = createDialogContent("Title"))
}

private fun createDialogContent(title: String) = @Composable {
   Text(title)
}
```

and because it looks like other Composable functions, it should behave the same way as:
```
fun MyComposable() {
   Box(content = createBoxContent("Title"))
}
```